tests/publiccloud/register_system.pm

```diff
@@ -34,14 +34,14 @@ sub run {
         # note: ssh_script_retry dies on failure
         $args->{my_instance}->retry_ssh_command("sudo SUSEConnect -r " . get_required_var('SCC_REGCODE'), timeout => 420, retry => 3);
 
+        # We need to pass the architecture of the target system, which is not set in /etc/os-release
+        my $arch = $args->{my_instance}->run_ssh_command(cmd => 'echo ${CPU}');
         for my $addon (@addons) {
             next if ($addon =~ /^\s+$/);
             if (is_sle('<15') && $addon =~ /tcm|wsm|contm|asmm|pcm/) {
-                ssh_add_suseconnect_product($args->{my_instance}->public_ip, get_addon_fullname($addon), '`echo ${VERSION} | cut -d- -f1`');
-            } elsif (is_sle('<15') && $addon =~ /sdk|we/) {
```

Comment:
and why you cutting sdk and we case ?

Reply:
I needed to add the `${VERSION_ID}` and `$arch` variable in both cases, which made them identical:

    } elsif (is_sle('<15') && $addon =~ /sdk|we/) {
        ssh_add_suseconnect_product($args->{my_instance}->public_ip, get_addon_fullname($addon), '${VERSION_ID}', $arch);
    } else {
        ssh_add_suseconnect_product($args->{my_instance}->public_ip, get_addon_fullname($addon), '${VERSION_ID}', $arch);
    }

So there was no need anymore for this case.

I did the [SLE 15-SP2 GCE-BYOS](http://duck-norris.qam.suse.de/t4468) verification run to check, if this breaks something. I add another one for a SLE12 variant, so we can be sure nothing breaks.

Reply:
in this case see  https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11705/files#r553363459